### ISSUE TITLE
remove attributes from highlight groups

### ISSUE THREAD

**Issue by GBeauregard** [neovim/neovim#16481]
### Feature already in Vim?

no

### Feature description

In [indent-blankline](https://github.com/lukas-reineke/indent-blankline.nvim) which uses virtual text to show line indents there's a tension between needing to suppress gui options that set italics/underline to avoid fudging the indention's virtual text while also wanting to allow mixing other gui elements that could have been set by the theme (f.ex. gui=reverse used for block selections).

This came up in this bug where the need to suppress the italics/bold conflicted with the theme: https://github.com/lukas-reineke/indent-blankline.nvim/issues/328

The [plugin author suggests](https://github.com/lukas-reineke/indent-blankline.nvim/issues/328#issuecomment-983345907) neovim could add negative gui options e.g. `noitalic` to resolve this tension and allow correctly expressing what indent-blankline is trying to accomplish without breaking theming.

> So a highlight group can set this, all gui options get combined, but `noitalic` overwrites if the underlying highlight group had `italic`

**Comment by clason** [comment]
This should be covered once `nvim_set_hl()` (which takes a `italic=false` key) can work with the global highlight namespace ("`ns=0`").

**Comment by zeertzjq** [comment]
I think OP doesn't want to modify existing highlight groups. OP wants higher-priority highlight groups to be able to disable single attributes from lower-priority highlight groups.

**Comment by clason** [comment]
Yes, but that's also covered by this? If the higher-priority group contains `italic=false`?

**Comment by zeertzjq** [comment]
That may depend on the exact implementation.

**Comment by ggandor** [comment]
This would also be immensely useful for all jumping plugins that highlight/label targets. `nvim_set_hl(0, 'Group', {<attr>=false}`) has no effect unfortunately.

**Comment by nickkadutskyi** [comment]
Keeping italic on lower priority group while removing it on a higher priority group or a more specific group would be useful. 
I am currently creating a color scheme and `:Inspect` returns this:
<pre><code>
Treesitter
  - @comment.lua links to LanguageDefaults_Comments_LineComment lua
  - @spell.lua links to @spell lua
  - <em>@comment.documentation.lua</em> links to <em>@comment.documentation.lua_Custom</em> lua

Semantic Tokens
  - @lsp.type.keyword.lua links to LanguageDefaults_Keyword priority: 125
  - @lsp.mod.documentation.lua links to Lua_Comments_EmmyDoc_Tag priority: 126
  - @lsp.typemod.keyword.documentation.lua links to @lsp.typemod.keyword.documentation.lua_Custom priority: 127
</code></pre>

I am styling a documentation comment and I want `@comment.documentation.lua` to italicize the text but `@lsp.typemod.keyword.documentation.lua` to make it normal so that I have italicized text while any keyword is having normal style within the doc comment.

**Comment by justinmk** [comment]
Does https://github.com/neovim/neovim/pull/37546 fix this?

**Comment by clason** [comment]
I don't think so; this issue is about how multiple extmarks interact, not with individual highlight groups.

**EDIT:** No it doesn't (unsurprisingly); I just tested to make sure.

To be clear, the ask here is for a variant of `italic = false` that is _not_ equivalent to not setting `italic` at all. I think the simplest solution would be to add a new attribute `upright` or `normal` that if set to `true` overrides  `italic` (and possibly `bold` and `underlined` etc. -- I don't think we need a `nofoo` variant for every attribute...)

**Comment by dietrichm** [comment]
This would also come in handy for text where both `DiagnosticDeprecated` and `DiagnosticUnderlineHint` are applied. One adds a strike-through and the other an underline. Both is visually noisy - just the strike-through would be enough.

```
Extmarks
  - DiagnosticDeprecated nvim.nvim.lsp.intelephense.1.diagnostic.underline
  - DiagnosticUnderlineHint nvim.nvim.lsp.intelephense.1.diagnostic.underline


DiagnosticDeprecated xxx cterm=strikethrough gui=strikethrough guisp=NvimLightRed
DiagnosticUnderlineHint xxx cterm=underline gui=underline guisp=NvimLightBlue
```

In other words just adding `upright` or `normal` would not fix this.
